Cpp2IL.Core/CorePlugin/CallAnalysisProcessingLayer.cs

```diff
@@ -0,0 +1,361 @@
+            var callsDeduplicatedMethodsAttributeInfo = callsDeduplicatedMethodsAttributes[assemblyAnalysisContext];
+            var callsAttributeInfo = callsAttributes[assemblyAnalysisContext];
+            var calledByAttributeInfo = calledByAttributes[assemblyAnalysisContext];
+
+            foreach (var m in assemblyAnalysisContext.Types.SelectMany(t => t.Methods))
+            {
+                if (m.CustomAttributes == null || m.Definition == null)
+                    continue;
+
+                AddOneParameterAttribute(m, callerCountAttributeInfo, GetCount(callCounts, m.UnderlyingPointer));
+                var unknownCallCount = GetCount(unknownCalls, m);
+                if (deduplicatedCalls.TryGetValue(m, out var deduplicatedCallCount))
+                {
+                    AddOneParameterAttribute(m, callsDeduplicatedMethodsAttributeInfo, deduplicatedCallCount);
+                }
+                if (callsDictionary.TryGetValue(m, out var callsList))
+                {
+                    foreach (var calledMethod in callsList)
+                    {
+                        var il2cppType = GetTypeFromContext(calledMethod.DeclaringType);
+                        if (il2cppType is not null)
+                        {
+                            AddTwoParameterAttribute(m, callsAttributeInfo, il2cppType, calledMethod.Name);
+                        }
+                        else
+                        {
+                            unknownCallCount++;
+                        }
+                    }
+                }
+                const int MaximumCalledByAttributes = 20;//We don't want 1000 attributes on a single method :)
+                if (calledByDictionary.TryGetValue(m, out var calledByList) && calledByList.Count < MaximumCalledByAttributes)
+                {
+                    foreach (var callingMethod in calledByList)
+                    {
+                        var il2cppType = GetTypeFromContext(callingMethod.DeclaringType);
+                        if (il2cppType is not null)
+                        {
+                            AddTwoParameterAttribute(m, calledByAttributeInfo, il2cppType, callingMethod.Name);
+                        }
+                        //If null, nothing we can do
+                    }
+                }
+                if (unknownCallCount > 0)
+                {
+                    AddOneParameterAttribute(m, callsUnknownMethodsAttributeInfo, unknownCallCount);
+                }
+            }
+        }
+    }
+
+    /// <summary>
+    /// Inject an attribute with no fields nor properties into the <see cref="ApplicationAnalysisContext"/>.
+    /// </summary>
+    /// <returns>A dictionary of assembly contexts to their inject attribute constructors.</returns>
+    private static Dictionary<AssemblyAnalysisContext, InjectedMethodAnalysisContext> InjectZeroParameterAttribute(
```

Comment:
Function parameters on one line please, regardless of how long that makes that line.